src/ansys/geometry/core/designer/design.py

```diff
@@ -818,17 +807,16 @@ def add_beam_circular_profile(
         if not dir_x.is_perpendicular_to(dir_y):
             raise ValueError("Direction X and direction Y must be perpendicular.")
 
-        request = CreateBeamCircularProfileRequest(
-            origin=point3d_to_grpc_point(center),
+        self._grpc_client.log.debug(f"Creating a beam circular profile on {self.id}...")
+
+        response = self._grpc_client._services.designs.create_beam_circular_profile(
+            center=center,
             radius=radius.value.m_as(DEFAULT_UNITS.SERVER_LENGTH),
```

Comment:
Pass in the ``radius`` object and use the conversion method inside the ``conversions.py`` module. There is one for transforming measurements. See other references